pkg/vpcmodel/abstractVPC.go

```diff
@@ -74,17 +74,40 @@ type NodeSet interface {
 	AddressRange() *common.IPBlock
 }
 
+type RulesInFilter struct {
+	// todo: is the assumption that the set of rules will always be kept in a list a valid one?
+	RulesIndexes int   // sg/nacl index in sgList/naclList
+	Rules        []int // list of indexes if rules in the sg/nacl
+
+}
+
+type rulesInLayer struct {
+	layer string
+	rules []RulesInFilter
+}
+
+type rulesInLayers []rulesInLayer
+
+type RulesOfConnection struct {
+	ingressRules rulesInLayers
+	egressRules  rulesInLayers
+}
+
 // FilterTrafficResource capture allowed traffic between 2 endpoints
 type FilterTrafficResource interface {
 	VPCResourceIntf
-	// get the connectivity result when the filterTraffic resource is applied to the given NodeSet element
+	// AllowedConnectivity get the connectivity result when the filterTraffic resource is applied to the given NodeSet element
```

Comment:
I only added the prefix `AllowedConnectivity` so that it will stop complaining, did not really spend time on that -  but indeed it seems a typo, I changes `NodeSet` to `Node`. Please verify...